### DISCUSSION TITLE
Unlocking VWZ MEH 97/6 for ebusd

### DISCUSSION THREAD

**Discussion by morphZ** [john30/ebusd#689] [Q&A]
Hi everyone,

I have a new Vaillant heat pump running at my home, comprising of the following components: 
- aroTHERM plus VWL 75/6
- VWZ MEH 97/6
- VRC 720
- esera usb
- ebusd via docker

Trying to get data out of it is giving me a hard time. After spending many hours into analyzing I have come up with the following conclusions:
- The VWZ (so called Hydraulikstation) is the main controller.
- The VWZ is not yet available in ebusd-configuration.
- I can't get any responds from the ebus after writing with the esera-usb, even though the arbitration is sucessful and I can see all bytes written to *and* received from the ebus in raw mode (e.g. `>31` and `<31`). This is already addressed in #677, but I'm not yet sure it is related to writing problems.

Could it be that the newer Vaillant components are deliberately not answering the scans and only answer to scans from Original Vaillant components?

I also read the #437 which is related because of the same aroTHERM being used. But even the config files there couldn't really decode more messages (only the WaterThroughput works)

Anyhow, now I've set my mind on decoding the VWZ 😄 

I would like to start with the messages, that I can already read from the bus. Which should somehow be the messages of the control loop between the VWZ and the VWL.

They basically look like this
```
7108b51a0304e732 / 0ee707c0cfc07b3c00b01b20000000 = 33
7108b51a0304e733 / 0ee70000c083020000000000000000 = 20
7108b51a0304e734 / 0ee7cf2000c0022101001800061800 = 37
7108b51a0304e735 / 0ee700000000000100900100000000 = 17
7108b51a0304e736 / 0ee700000000000000000000000000 = 33
7108b51a0304e832 / 0ee807c0cfc07b3c00b01b20000000 = 17
7108b51a0304e833 / 0ee80000c083020000000000000000 = 33
7108b51a0304e834 / 0ee8cf2000c0022101001800061800 = 20
7108b51a0304e835 / 0ee800000000000100900100000000 = 37
7108b51a0304e836 / 0ee800000000000000000000000000 = 17
```

What I can see so far:
- PBSB is b51a
- The first byte of the master data is always `0x04`
- The second byte of the master data is a counter going from `0x00` to `0xff` (in the above snippet `0xe7` and `0xe8`)
- The counter byte is always repeated by the slave in the first byte of the slave message
- After `0xff` it resets to `0x00`
- The last byte of the master data seems to be a subcommand and goes from `0x32` to `0x36`.
- The `0x36` subcommand always yields only zeros in the slave's answer.

Any advice on how to continue from here?
How can I best decode the messages? (I already know about and tried `ebusctl grab result decode` but due to the large quantity of different messages (because of the counter byte) it was no real help yet.
Has anyone managed to decode databytes via a python script? That would help me alot, but I couldn't get it to run (already tried [pyebus](https://pypi.org/project/pyebus/) without success)

Thanks in advance
morphZ

**Comment by cyberthom42** [comment]
Ich have got a similar setup (VWL 55/6 , VWZ MEH 97/6, VRC 720f, VPM W Trinkwasserstation). 
So far I was under the impression that the hmu delivers also the ebus data from the VWZ. At least I can read all sensor information and most of the hmu contents. In addition I can control the MEH using 720 ebus codes. 
But I also see the messages that you described. It would be great if we can decode them as well. 
However, what are you expecting to get out of this? The controller is the 720/720f. You can check this: You should not see the "Heizbetrieb" icon above the right selection key. This proves that the 720 is the controller.
